ros-ocp/templates/envoy-config-ingress.yaml

```diff
@@ -0,0 +1,196 @@
+                      if account_number == "" then
+                        account_number = org_id  -- Fallback to org_id
+                      end
+
+                      -- Build and encode X-Rh-Identity header
+                      local xrhid_json = build_xrhid_json(org_id, account_number, "User")
+                      local xrhid_b64 = request_handle:base64Escape(xrhid_json)
+
+                      -- Inject X-Rh-Identity header (for standard identity-based systems)
+                      request_handle:headers():add("X-Rh-Identity", xrhid_b64)
+
+                      -- Inject headers for insights-ros-ingress app (expects JWT token directly)
+                      request_handle:headers():add("X-ROS-Authenticated", "true")
+                      -- Forward the original Authorization header as X-Bearer-Token
+                      local auth_header = request_handle:headers():get("authorization")
+                      if auth_header then
+                        -- Extract Bearer token from Authorization header
+                        local bearer_token = string.match(auth_header, "^Bearer%s+(.+)$")
+                        if bearer_token then
+                          request_handle:headers():add("X-Bearer-Token", bearer_token)
+                        end
+                      end
+
+                      -- Log for debugging
+                      request_handle:logInfo(string.format("JWT Authentication successful - org_id: %s, account: %s",
+                                                           org_id, account_number))
+                    end
+
+              - name: envoy.filters.http.router
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+
+      clusters:
+      # Ingress backend (port 8081 when JWT enabled)
+      - name: ros-ingress-backend
+        connect_timeout: 5s
+        type: STATIC
+        load_assignment:
+          cluster_name: ros-ingress-backend
+          endpoints:
+          - lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: 127.0.0.1
+                    port_value: 8081
+
+      # Keycloak JWKS endpoint for JWT signature verification
+      - name: keycloak_jwks
+        connect_timeout: 5s
+        type: STRICT_DNS
+        load_assignment:
+          cluster_name: keycloak_jwks
+          endpoints:
+          - lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: {{ $keycloakUrl | replace "https://" "" | replace "http://" "" }}
+                    port_value: 443
+        transport_socket:
+          name: envoy.transport_sockets.tls
+          typed_config:
+            "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
+            sni: {{ $keycloakUrl | replace "https://" "" | replace "http://" "" }}
```

Comment:
do you think we should add a `validation_context`? without it will envoy accepts any certificate from the Keycloak server.

Reply:
Done!